src/main/groovy/netflix/nebula/dependency/recommender/provider/FuzzyVersionResolver.java

```diff
@@ -42,6 +40,16 @@ public String versionOf(String key) {
         return null;
     }
 
+    private void initializeGlobCache() {
```

Comment:
Why not leave this inline? At least, probably better to have this return the list rather than initializing as a side effect.

Reply:
Actually, it _must_ return the completed list rather than initializing in-place. Writing the `globs` reference under lock before the list is fully initialized exposes it to access from outside prematurely.